lib/plugins/aws/provider/awsProvider.js

```diff
@@ -89,19 +89,31 @@ const impl = {
         });
       };
 
-      const profileCredentials = new AWS.SharedIniFileCredentials(params);
-      if (
-        !(
-          profileCredentials.accessKeyId ||
-          profileCredentials.sessionToken ||
-          profileCredentials.roleArn
-        )
-      ) {
-        throw new Error(`Profile ${profile} does not exist`);
-      }
-
-      impl.addCredentials(results, profileCredentials);
+      const profileCredentials = [
+        BbPromise.resolve(new AWS.SharedIniFileCredentials(params)),
```

Comment:
I would avoid using `BbPromis.resolve`, there's no real use case here for that

If function is promise returning, I would wrap it's body, with `BbPromise.try`, and just use regular `return` and `throw` inside. It'll make future async/await migration very straightforward (which can be just about removing the wrap and adding `async` in front of a function)